graphql/mutation_resolver.go

```diff
@@ -108,6 +108,37 @@ func (r *mutationResolver) RemoveAnnotationIssue(ctx context.Context, taskID str
 	}
 }
 
+// AddAnnotationMetadataLink is the resolver for the addAnnotationMetadataLink field.
+func (r *mutationResolver) AddAnnotationMetadataLink(ctx context.Context, taskID string, execution int, apiMetadataLink restModel.APIMetadataLink) (bool, error) {
+	metadataLink := restModel.APIMetadataLinkToService(apiMetadataLink)
+	if err := annotations.ValidateMetadataLinks(*metadataLink); err != nil {
+		return false, InputValidationError.Send(ctx, fmt.Sprintf("invalid metadata link: %s", err.Error()))
+	}
+	annotation, err := annotations.FindOneByTaskIdAndExecution(taskID, execution)
+	if err != nil {
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("finding annotation: %s", err.Error()))
+	}
+	if annotation == nil {
```

Comment:
Should this be an error? What if there isn't an annotation yet, wouldn't we just want to add one? (Similar to how the add issue mutations work)